Services/communication_service.c

```diff
@@ -35,31 +53,753 @@
+        U_stat.set.PIPE_t = csp_ntoh16(U_stat.set.PIPE_t);
+        cnv8_16LE(&packet->data[IN_DATA_BYTE + 6],
+                &U_stat.set.beacon_t);
+        U_stat.set.beacon_t = csp_ntoh16(U_stat.set.beacon_t);
+        cnv8_16LE(&packet->data[IN_DATA_BYTE + 8],
+                &U_stat.set.audio_t);
+        U_stat.set.audio_t = csp_ntoh16(U_stat.set.audio_t);
+        status =  HAL_UHF_setFreq(U_stat.set.freq)
+                + HAL_UHF_setPIPEt(U_stat.set.PIPE_t)
+                + HAL_UHF_setBeaconT(U_stat.set.beacon_t)
+                + HAL_UHF_setAudioT(U_stat.set.audio_t);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_RESTORE_DEFAULT:
+        status = HAL_UHF_restore(&U_restore);
+        if (sizeof(U_restore) + 1 > csp_buffer_data_size())
+        {
+            return SATR_ERROR;
+        }
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &U_restore, sizeof(U_restore));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(U_restore) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_LOW_PWR:
+        status = HAL_UHF_lowPwr(&U_stat.low_pwr_stat);
+        if (sizeof(U_stat.low_pwr_stat) + 1 > csp_buffer_data_size())
+        {
+            return SATR_ERROR;
+        }
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &U_stat.low_pwr_stat,
+               sizeof(U_stat.low_pwr_stat));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(U_stat.low_pwr_stat) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_SET_DEST:
+        //uhf_struct_len = (uint8_t) packet->data[IN_DATA_BYTE]; //If using this, add +1 in the loop
+        uhf_struct_len = 6;
+        U_callsign.dest.len = uhf_struct_len;
+        for (i = 0; i < uhf_struct_len; i++)
+                {
+                    U_callsign.dest.message[i] =
+                            (uint8_t) packet->data[IN_DATA_BYTE + 3 + 4*i];
+                    U_callsign.dest.message[i] = csp_ntoh32(
+                            (uint32_t) U_callsign.dest.message[i]);
+                }
+        status = HAL_UHF_setDestination(U_callsign.dest);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_SET_SRC:
+        //uhf_struct_len = (uint8_t) packet->data[IN_DATA_BYTE]; //If using this, add +1 in the loop
+        uhf_struct_len = 6;
```

Comment:
Same with this, and the ones below - can we do a `sizeof()` call instead?

Reply:
The easy way is to use the commented-out line but it requires the ground user to specify the length of the string which doesn't make sense unless for the beacon message. The alternative would be to define them as macros for example if we want to keep them tidy in some place if that's what you're worried about.

Reply:
Yeah, definately they should be defined at the top or in the header if we go the route of having these as fixed length strings, we should also make sure to make a comment here, and in the ground station that that is the case.
I wouldn't mind there being a size feild in the ground station command if that might help us -- what does this actually do? Just set the callsign? If yes, then I guess we should just do `#define CALLSIGN_LEN 6` or something just to make sure it's clear what this number represents

Reply:
And do the same everywhere that that happens, like `#define BEACON_LEN 32` or whatever